ctl/backup.go

```diff
@@ -62,26 +60,61 @@ func (cmd *BackupCommand) Run(ctx context.Context) error {
 		return err
 	}
 
-	// Open output file.
-	f, err := os.Create(cmd.Path)
+	schema, err := client.Schema(ctx)
 	if err != nil {
 		return err
 	}
-	defer f.Close()
 
-	// Begin streaming backup.
-	if err := client.BackupTo(ctx, f, cmd.Index, cmd.Frame, cmd.View); err != nil {
-		return err
+	var w = cmd.Stdout
+	if cmd.Path != "" {
+
+		filename := cmd.Path + ".pak"
+		log.Println("Creating:", filename)
+		f, err := os.Create(filename)
+		defer f.Close()
+		if err != nil {
+			return err
+		}
+		w = f
 	}
-
-	// Sync & close file to ensure durability.
-	if err := f.Sync(); err != nil {
-		return err
-	} else if err = f.Close(); err != nil {
+	gw := gzip.NewWriter(w)
+	tw := tar.NewWriter(gw)
+	//need to encode schema
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
```

Comment:
what do you think about using protobuf to encode the schema? the `cluster-resize` branch already has a proto `Schema` message defined.